client/src/app/users/user-filter/user-filter.component.ts

```diff
@@ -1,40 +1,34 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
 import { MatSnackBar } from '@angular/material/snack-bar';
 import { Subject, takeUntil } from 'rxjs';
-import { User, UserRole } from './user';
-import { UserService } from './user.service';
+import { User, UserRole } from '../user';
+import { UserService } from '../user.service';
 
 /**
- * A component that displays a list of users, either as a grid
- * of cards or as a vertical list.
- *
- * The component supports local filtering by name and/or company,
- * and remote filtering (i.e., filtering by the server) by
- * role and/or age. These choices are fairly arbitrary here,
+ * The component supports:
+ *  - local filtering (i.e., filtering by the client) by name and/or company,
+ *  - remote filtering (i.e., filtering by the server) by role and/or age.
+ * These choices are fairly arbitrary here,
  * but in "real" projects you want to think about where it
- * makes the most sense to do the filtering.
+ * makes the most sense to do each kind of filtering.
  */
 @Component({
-  selector: 'app-user-list-component',
-  templateUrl: 'user-list.component.html',
-  styleUrls: ['./user-list.component.scss'],
-  providers: []
+  selector: 'app-user-filter',
+  templateUrl: './user-filter.component.html',
+  styleUrls: ['./user-filter.component.scss']
 })
 
-export class UserListComponent implements OnInit, OnDestroy  {
-  // These are public so that tests can reference them (.spec.ts)
-  public serverFilteredUsers: User[];
-  public filteredUsers: User[];
-
-  public userName: string;
-  public userAge: number;
-  public userRole: UserRole;
-  public userCompany: string;
-  public viewType: 'card' | 'list' = 'card';
+export class UserFilterComponent implements OnInit, OnDestroy {
+  @Output() readonly filteredUsersChange = new EventEmitter<User[]>();
+  @Input() filteredUsers: User[];
```

Comment:
Making these into input and output and stating where to emit changes (line 93) keeps the filteredUsers up-to-date.

Reply:
I'm not sure if this was the case when you posted your comment, but we use `@Input` in UserCardComponent. One question that @floogulinc asked in UserListComponent is why use an `@Input` there since the UserListComponent is not embedded anywhere. This is in contrast with the UserCardComponent, which is used in the card view of the list and in the profile view of a single user. So, maybe this is an important distinction I didn't know about. Perhaps I was imposing a parent/child relationship between the filter component and the list component?